### ISSUE TITLE
react-native-maps 8.2 crashes with react-native v0.34.1 on Android physical device + simulator

### ISSUE THREAD

**Issue by comountainclimber** [airbnb/react-native-maps#647]
My application crashes anytime the mapview component is rendered on android with any version higher than 0.31 of react native.

dependencies are as follows:
    `"react": "~15.3.0",
    "react-native": "^0.34.1",
    "react-native-maps": "^0.8.2",`

Error in the android studio:
![group](https://cloud.githubusercontent.com/assets/13072035/19084535/32bb065c-8a24-11e6-91ac-a3a3f2f309cc.png)

`java.lang.NoSuchMethodError: No direct method <init>(IJ)V in class Lcom/facebook/react/uimanager/events/Event; or its super classes (declaration of 'com.facebook.react.uimanager.events.Event' appears in /data/data/com.onsitemobile/files/instant-run/dex/slice-com.facebook.react-react-native-0.34.1_2f4ce034c8d533589ff217c6e68bebefd11943b6-classes.dex)`

**Comment by AntiHate** [comment]
+1

**Comment by comountainclimber** [comment]
super confusing that no one else seems to have this issue...

**Comment by huangkaiw3n** [comment]
Same here, crashing on device Xperia Z5 and Samsung Note 4 on Android 6.

**Comment by huangkaiw3n** [comment]
I solved my issue. It was due to the project having build.gradle files with different versions of `com.google.android.gms:play-services`.  
There was a library that used the latest version which was 
`compile 'com.google.android.gms:play-services-gcm:+`, whereas react-native-maps used specifically 9.4.0, where its build.gradle contained,
`compile "com.google.android.gms:play-services-base:9.4.0"
  compile 'com.google.android.gms:play-services-maps:9.4.0'`.
After changing the offending library's version to use 9.4.0 instead of the lastest, the issue was solved.
